integration-libs/punchout/core/facade/punchout.service.ts

```diff
@@ -127,6 +130,20 @@ export class PunchoutService implements PunchoutFacade {
       }),
       take(1),
       switchMap((punchoutState: PunchoutState) => {
+        // prevent manual navigation to requisition page
+        if (
+          punchoutState?.closePunchoutSession === undefined &&
+          punchoutState?.cancelRequisition === undefined
+        ) {
```

Comment:
I didn't add new properties, just observed possible condition can catch an event - is there any other edge case that I don't know, and it causes this condition can't be used here?

Reply:
All use cases works fine for me. great! i haven't thought that boolean can actually have 3 values (undefined,true,false), so no need to add more properties.